### PR TITLE
#5994 - Refactor/rename TypeScript types Slice 4

### PR DESCRIPTION
**Pull request by grahamlangford**
## What does this PR do?

- Part of #5994

## Reviewer Tips
- see `src/types/extensionTypes.ts` for type name changes that weren't in the spec

## Discussion
- covers some final name changes
- includes typeguard name changes

## Future Work
- Rename types files to better fit new types
- Rename factory functions

## Checklist

- [ ] Designate a primary reviewer

### CODE REVIEW

File: src/types/extensionTypes.ts [hunk 1]
```diff
@@ -211,14 +212,14 @@ export type ActivatedModComponent<
  * An `ModComponentBase` with all inner definitions resolved.
  * @see resolveDefinitions
  */
-export type ResolvedExtension<Config extends UnknownObject = UnknownObject> =
+export type ResolvedModComponent<Config extends UnknownObject = UnknownObject> =
```
Comment (grahamlangford) [inline src/types/extensionTypes.ts:215]: Calling out type name change that wasn't in the spec.

File: src/types/extensionTypes.ts [hunk 2]
```diff
@@ -232,27 +233,27 @@ export type ResolvedExtension<Config extends UnknownObject = UnknownObject> =
 /**
  * A reference to an ModComponentBase.
  */
-export type ExtensionRef = {
+export type ModComponentRef = {
```
Comment (grahamlangford) [inline src/types/extensionTypes.ts:236]: Calling out type name change that wasn't in the spec.

File: src/types/extensionTypes.ts [hunk 3]
```diff
@@ -153,42 +153,43 @@ export type ModComponentBase<Config extends UnknownObject = UnknownObject> = {
...
  *
  * @see ModComponentBase
- * @see ResolvedExtension
+ * @see ResolvedModComponent
  */
-export type UnresolvedExtension<Config extends UnknownObject = UnknownObject> =
-  ModComponentBase<Config> & {
-    _unresolvedExtensionBrand: never;
-  };
+export type UnresolvedModComponent<
```
Comment (grahamlangford) [inline src/types/extensionTypes.ts:177]: Calling out type name change that wasn't in the spec.

File: src/extensionConsole/pages/mods/hooks/useViewPublishAction.ts [hunk 4]
```diff
@@ -45,7 +49,7 @@ function useViewPublishAction(modViewItem: ModViewItem): () => void | null {
     // Deployment sharing is controlled via the Admin Console
     !isDeployment &&
     // Extensions can be published
-    (isExtension(mod) ||
+    (isResolvedExtension(mod) ||
```
Comment (twschiller) [inline src/extensionConsole/pages/mods/hooks/useViewPublishAction.ts:52]: Should this be isResolvedModComponent?

File: src/types/extensionTypes.ts (hunk 1 above)
Comment (twschiller) [reply src/types/extensionTypes.ts:215]: @grahamlangford I think we want to call this HydratedModComponent. We've decided to use hydrated vs. resolved to indicated if inner definitions have been hydrated

File: src/extensionConsole/pages/mods/hooks/useViewPublishAction.ts (hunk 4 above)
Comment (grahamlangford) [reply src/extensionConsole/pages/mods/hooks/useViewPublishAction.ts:52]: You're right. I'm going through and finding more extension -> ModComponent and extension point -> StarterBrick changes right now.

**Review by twschiller — APPROVED** [review]

**Pull request merged**